### ISSUE TITLE
Markup and a11y mappings for navigation menus

### ISSUE THREAD

**Issue by josepharhar** [openui/open-ui#1452]
For the next iteration of menu elements which are for links instead of menuitems, what should the a11y mappings be and how does that constrain what the HTML markup can be? There may have been discussions about this already somewhere.

It seems like there are two relevant APG examples:
- Navigation menubar: https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/
- Disclosure navigation menu: https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/

Should each "item" which opens a submenu be mapped as a button or a menuitem? Should each list of links and buttons which open submenus be mapped as a menu or as a list?

The disclosure navigation menu wraps all of the markup in one nav element, but the structure we are currently going for with menulist/menubar/menuitem doesn't use nesting to put the entire thing in one element. Can we still avoid nesting for the HTML to keep it consistent with application menus and navigation menus?

@scottaohara

**Comment by josepharhar** [comment]
Was there an actual discussion about this? the comment from css-meeting-bot only shows one message in irc

**Comment by dbaron** [comment]
No, we decided to wait until you were around. :-)

**Comment by josepharhar** [comment]
For the HTML markup, what if we renamed menubar to navbar and menulist to navlist? I'm not sure how to represent nested submenus in a way that avoids the menuitem element entirely, which seems like it should be a goal in order to prevent having menuitems which aren't links inside a navigation menu. Maybe that should be a new element too?

```html
<navbar>
  <submenu>
    <menuitem>fruits</menuitem>
    <navlist>
      <a href="orange.html">orange</a>
      <a href="apple.html">apple</a>
    </navlist>
  </submenu>
  <submenu>
    <menuitem>vegetables</menuitem>
    <navlist>
      <a href="potato.html">potato</a>
      <a href="carrot.html">carrot</a>
    </navlist>
  </submenu>
</navbar>
```

In other words:
- `<menubar>` -> `<navbar>`
- `<menulist>` -> `<navlist>`
- `<menuitem>` which opens a submenu is still `<menuitem>`
- `<menuitem>` which does an action -> `<a>`

I figured that we need menulist to have a new tag name in the case that the author creates a menu popup without an in-page menubar for a navigation menu.

I'm still not sure what the a11y mappings of these elements should be.

**Comment by josepharhar** [comment]
some notes from the meeting:
- we probably shouldn't look at apg for guidance on this
- behaviors for navigation menus are probably supposed to be different than application menus
  - tab should go through links instead of closing the popover
  - arrow keys should not move focus like they do in application menus
- we should enumerate the differences between application and navigation menus
  - is the mouse/pointer behavior any different?
